### ISSUE TITLE
Investigate possible bug with algod.net creation when enabling catchpoint tracking

### ISSUE THREAD

**Issue by onetechnical** [algorand/go-algorand#4774]
### Subject of the issue

When the user makes a change in the config.json and restarts the services , it fails completely.
- algod.net file not created
- goal node status returns "connection refused"

Occurs when CatchpointTracking is changed from -1 to 0 or 1

config being used

```
{
    "AnnounceParticipationKey": false,
    "Archival": true,
    "BaseLoggerDebugLevel": 3,
    "CatchpointTracking": -1,
    "DeadlockDetection": 1,
    "EnableBlockService": true,
    "EnableLedgerService": true,
    "EnableMetricReporting": true,
    "EndpointAddress": "0.0.0.0:8081",
    "FallbackDNSResolverAddress": "1.1.1.1",
    "NetAddress": ":4160",
    "PeerPingPeriodSeconds": 60,
    "PublicAddress": "REDACTED:80",
    "TelemetryToLog": false
}
```
Which
```
$ sudo systemctl restart algorand (OK)
$ sudo systemctl status algorand (OK)
$ goal node -w 500 status (OK)
```

but when changed
```
{
    "AnnounceParticipationKey": false,
    "Archival": true,
    "BaseLoggerDebugLevel": 3,
    "CatchpointTracking": 1,
    "DeadlockDetection": 1,
    "EnableBlockService": true,
    "EnableLedgerService": true,
    "EnableMetricReporting": true,
    "EndpointAddress": "0.0.0.0:8081",
    "FallbackDNSResolverAddress": "1.1.1.1",
    "NetAddress": ":4160",
    "PeerPingPeriodSeconds": 60,
    "PublicAddress": "REDACTED:80",
    "TelemetryToLog": false
}
```
returns
```
$ sudo systemctl restart algorand (long delay, OK)
$ sudo systemctl status algorand (OK)
$ goal node -w 500 status (Cannot contact Algorand node: open /var/lib/algorand/algod.net: no such file or directory)
```

No extra output in node.log even with BaseLoggerDebugLevel set to 5

### Steps to reproduce

1. Try starting node with first config
2. Change config to second
3. Restart node, run goal node status

### Expected behaviour

Standard status output

### Actual behaviour

Error message above returned.

**Comment by tom-gnade** [comment]
Hello, I am the original reporter of this issue. I've been requested to add the following information:

Operating System: Ubuntu 20.04.5 LTS (Focal Fossa)
Install Method: apt (standard method for Ubuntu https://developer.algorand.org/docs/run-a-node/setup/install/)
Algod version: 3.11.2.stable [rel/stable] (commit #99b37ac0)

Permissions in /var/lib/algorand directory:

755 drwxr-xr-x 6 algorand algorand 23 Nov 18 05:09 ./
666 -rw-rw-rw- 1 algorand algorand 9125101 Nov  9 00:39 agreement.cdv
666 -rw-rw-rw- 1 algorand algorand 10738079 Nov  9 00:39 agreement.cdv.archive
644 -rw-r--r-- 1 algorand algorand 64 Oct 25 10:34 algod.admin.token
644 -rw-r--r-- 1 algorand algorand 17 Nov 17 05:49 algod-listen.net
600 -rw------- 1 algorand algorand 0 Oct 25 10:34 algod.lock
644 -rw-r--r-- 1 algorand algorand 10 Nov 17 05:49 algod.net
644 -rw-r--r-- 1 algorand algorand 8 Nov 17 05:49 algod.pid
644 -rw-r--r-- 1 algorand algorand 64 Oct 25 10:34 algod.token
755 drwxr-xr-x 2 algorand algorand 3 Oct 27 10:14 .algorand/
644 -rw-r--r-- 1 algorand algorand 3962 Nov 17 05:49 config.json
644 -rw-r--r-- 1 algorand algorand 3906 Oct 22 05:12 config.json.example
775 drwxrwxr-x 7 algorand algorand 7 Oct 25 10:34 genesis/
644 -rw-r--r-- 1 algorand algorand 24972 Oct 22 05:12 genesis.json
775 drwxrwxr-x 3 algorand algorand 3 Oct 25 10:37 .local/
644 -rw-r--r-- 1 algorand algorand 210 Oct 27 10:14 logging.config
755 drwxr-xr-x 2 algorand algorand 17 Nov 17 05:49 mainnet-v1.0/
644 -rw-r--r-- 1 algorand algorand 355224271 Nov 19 02:38 node.log
644 -rw-r--r-- 1 algorand algorand 58 Oct 22 05:12 system.json

**Comment by onetechnical** [comment]
I think I've got something. I was missing the detail of it being fully caught up. Ran the following on focal fossa:

- Install from debian package
- Run fast catchup (with defaults)
- Change the following config setting: `"CatchpointTracking": 1,`
- Run a `sudo systemctl restart algorand`

This will start accountsInitialize:

```
{"file":"catchpointtracker.go","function":"github.com/algorand/go-algorand/ledger.(*catchpointTracker).accountsInitializeHashes","level":"info","line":1461,"msg":"accountsInitialize rebuilding merkle trie for round 24944374","name":"","time":"2022-11-18T21:01:57.283114Z"}
```

This will then fail with the following:

```
{"file":"catchpointtracker.go","function":"github.com/algorand/go-algorand/ledger.(*catchpointTracker).accountsInitializeHashes","level":"info","line":1518,"msg":"accountsInitialize still building the trie, and hashed so far 14630912 accounts","name":"","time":"2022-11-18T21:01:38.459771Z"}
{"callee":"github.com/algorand/go-algorand/ledger.(*catchpointTracker).loadFromDisk.func1","caller":"github.com/algorand/go-algorand/ledger/catchpointtracker.go:342","file":"dbutil.go","function":"github.com/algorand/go-algorand/util/db.(*Accessor).atomic","level":"warning","line":307,"msg":"dbatomic: tx surpassed expected deadline by 3m37.849728876s","name":"","readonly":false,"time":"2022-11-18T21:01:51.336636Z"}
{"file":"node.go","function":"github.com/algorand/go-algorand/node.MakeFull","level":"error","line":215,"msg":"[Stack] goroutine 1 [running]:\nruntime/debug.Stack()\n\truntime/debug/stack.go:24 +0x65\ngithub.com/algorand/go-algorand/logging.logger.Errorf({0xc00012c1c0, 0xc0000104a0}, {0x12887d1, 0x21}, {0xc00e77ea40, 0x2, 0x2})\n\tgithub.com/algorand/go-algorand/logging/log.go:229 +0x69\ngithub.com/algorand/go-algorand/node.MakeFull({_, _}, {_, _}, {0x18, 0x0, 0x4, {0x0, 0x0}, 0xdf8475800, ...}, ...)\n\tgithub.com/algorand/go-algorand/node/node.go:215 +0xc5d\ngithub.com/algorand/go-algorand/daemon/algod.(*Server).Initialize(_, {0x18, 0x0, 0x4, {0x0, 0x0}, 0xdf8475800, {0x0, 0x0}, 0x1e, ...}, ...)\n\tgithub.com/algorand/go-algorand/daemon/algod/server.go:174 +0xcf3\nmain.run()\n\tgithub.com/algorand/go-algorand/cmd/algod/main.go:299 +0x1d1e\nmain.main()\n\tgithub.com/algorand/go-algorand/cmd/algod/main.go:61 +0x5c\n","time":"2022-11-18T21:01:52.068628Z"}
{"file":"node.go","function":"github.com/algorand/go-algorand/node.MakeFull","level":"error","line":215,"msg":"Cannot initialize ledger (/var/lib/algorand/mainnet-v1.0/ledger): reloadLedger.loadFromDisk tracker *ledger.catchpointTracker failed to loadFromDisk : resource table entries exceed the ones specified in the accountbase table","time":"2022-11-18T21:01:52.068693Z"}
{"file":"main.go","function":"main.run","level":"error","line":302,"msg":"[Stack] goroutine 1 [running]:\nruntime/debug.Stack()\n\truntime/debug/stack.go:24 +0x65\ngithub.com/algorand/go-algorand/logging.logger.Error({0xc00012c1c0, 0xc0000104a0}, {0xc00e770bf0, 0x1, 0x1})\n\tgithub.com/algorand/go-algorand/logging/log.go:219 +0x53\nmain.run()\n\tgithub.com/algorand/go-algorand/cmd/algod/main.go:302 +0x2106\nmain.main()\n\tgithub.com/algorand/go-algorand/cmd/algod/main.go:61 +0x5c\n","time":"2022-11-18T21:01:52.068747Z"}
{"file":"main.go","function":"main.run","level":"error","line":302,"msg":"couldn't initialize the node: reloadLedger.loadFromDisk tracker *ledger.catchpointTracker failed to loadFromDisk : resource table entries exceed the ones specified in the accountbase table","time":"2022-11-18T21:01:52.068775Z"}
```

**Comment by onetechnical** [comment]
[node.log.gz](https://github.com/algorand/go-algorand/files/10044905/node.log.gz)

**Comment by onetechnical** [comment]
From syslog:

```
Nov 18 21:01:52 ip-172-31-77-143 algod[5410]: couldn't initialize the node: reloadLedger.loadFromDisk tracker *ledger.catchpointTracker failed to loadFromDisk : resource table entries exceed the ones specified in the accountbase table
Nov 18 21:01:57 ip-172-31-77-143 algod[5446]: Telemetry configured from '/var/lib/algorand/logging.config'
Nov 18 21:01:57 ip-172-31-77-143 algod[5446]: Logging to:  /var/lib/algorand/node.log
Nov 18 21:05:13 ip-172-31-77-143 algod[5446]: couldn't initialize the node: reloadLedger.loadFromDisk tracker *ledger.catchpointTracker failed to loadFromDisk : resource table entries exceed the ones specified in the accountbase table
```

**Comment by tom-gnade** [comment]
I can check on my relay to see if I get that same error message in the syslog if you'd like.

**Comment by onetechnical** [comment]
Can you get us the full node.log @tom-gnade ?

**Comment by tom-gnade** [comment]
$ sudo systemctl restart algorand
$ sudo systemctl status algorand

*** SERVICE STATUS "CatchpointTracking": -1 *******
* algorand.service - Algorand daemon for mainnet in /var/lib/algorand
     Loaded: loaded (/lib/systemd/system/algorand.service; enabled; vendor preset: enabled)
    Drop-In: /etc/systemd/system/algorand.service.d
             `-override.conf
     Active: active (running) since Tue 2022-11-22 14:20:21 UTC; 6s ago
   Main PID: 138142 (algod)
      Tasks: 23 (limit: 76946)
     Memory: 246.2M
     CGroup: /system.slice/algorand.service
             |-138142 /usr/bin/algod -d /var/lib/algorand
             `-138169 ./node_exporter --web.listen-address=:9100 --web.telemetry-path=/metrics

Nov 22 14:20:21 systemd[1]: Started Algorand daemon for mainnet in /var/lib/algorand.
Nov 22 14:20:21 algod[138142]: Telemetry configured from '/var/lib/algorand/logging.config'
Nov 22 14:20:21 algod[138142]: Logging to:  /var/lib/algorand/node.log
Nov 22 14:20:22 algod[138142]: Deadlock detection is set to: enabled (Default state is 'disable')
Nov 22 14:20:22 algod[138142]: Initializing the Algorand node... Success!
Nov 22 14:20:22 algod[138142]: ▒<87>▒ http server started on [::]:8081
Nov 22 14:20:22 algod[138142]: Node running and accepting RPC requests over HTTP on port [::]:8081>
*** END SERVICE STATUS "CatchpointTracking": -1 *******

$ sudo systemctl restart algorand
$ sudo systemctl status algorand

*** SERVICE STATUS "CatchpointTracking": 1 *******
* algorand.service - Algorand daemon for mainnet in /var/lib/algorand
     Loaded: loaded (/lib/systemd/system/algorand.service; enabled; vendor preset: enabled)
    Drop-In: /etc/systemd/system/algorand.service.d
             `-override.conf
     Active: active (running) since Tue 2022-11-22 14:41:41 UTC; 2s ago
   Main PID: 146619 (algod)
      Tasks: 15 (limit: 76946)
     Memory: 611.5M
     CGroup: /system.slice/algorand.service
             `-146619 /usr/bin/algod -d /var/lib/algorand

Nov 22 14:41:41 systemd[1]: Started Algorand daemon for mainnet in /var/lib/algorand.
Nov 22 14:41:41 algod[146619]: Telemetry configured from '/var/lib/algorand/logging.config'
Nov 22 14:41:41 algod[146619]: Logging to:  /var/lib/algorand/node.log
*** END SERVICE STATUS "CatchpointTracking": 1 *******

NOTE: CGroup the node_exporter/metrics process has not spawned.
      Also, the "Deadlock detection" message is not displayed - so the startup failed immediately after parsing config.json 

$ journalctl --since "2022-11-22 14:20:15"

*** SYSLOG STARTING FROM 2022-11-22 14:20:15 (filtered to remove irrelevant entries) *******
-- Logs begin at Tue 2022-09-13 14:44:56 UTC, end at Tue 2022-11-22 14:40:02 UTC. --
Nov 22 14:20:21 systemd[1]: algorand.service: Succeeded.
Nov 22 14:20:21 systemd[1]: Stopped Algorand daemon for mainnet in /var/lib/algorand.
Nov 22 14:20:21 systemd[1]: Started Algorand daemon for mainnet in /var/lib/algorand.
Nov 22 14:20:21 sudo[138062]: pam_unix(sudo:session): session closed for user root
Nov 22 14:20:21 algod[138142]: Telemetry configured from '/var/lib/algorand/logging.config'
Nov 22 14:20:21 algod[138142]: Logging to:  /var/lib/algorand/node.log
Nov 22 14:20:22 algod[138142]: Deadlock detection is set to: enabled (Default state is 'disable')
Nov 22 14:20:22 algod[138142]: Initializing the Algorand node... Success!
Nov 22 14:20:22 algod[138142]: ⇨ http server started on [::]:8081
Nov 22 14:20:22 algod[138142]: Node running and accepting RPC requests over HTTP on port [::]:8081. Press Ctrl-C to exit
Nov 22 14:20:28 sudo[138178]:     root : TTY=pts/0 ; PWD=/var/lib/algorand ; USER=root ; COMMAND=/usr/bin/systemctl status algorand
Nov 22 14:20:28 sudo[138178]: pam_unix(sudo:session): session opened for user root by root(uid=0)
Nov 22 14:21:01 sudo[138178]: pam_unix(sudo:session): session closed for user root
Nov 22 14:22:24 psad[813]: src: 132.148.74.157 signature match: "MISC Microsoft SQL Server communication attempt" (sid: 100205) tcp port: 1433
Nov 22 14:25:01 CRON[139934]: pam_unix(cron:session): session opened for user root by (uid=0)
Nov 22 14:25:01 CRON[139935]: (root) CMD (command -v debian-sa1 > /dev/null && debian-sa1 1 1)
Nov 22 14:25:01 CRON[139934]: pam_unix(cron:session): session closed for user root
Nov 22 14:35:01 CRON[143840]: pam_unix(cron:session): session opened for user root by (uid=0)
Nov 22 14:35:01 CRON[143841]: (root) CMD (command -v debian-sa1 > /dev/null && debian-sa1 1 1)
Nov 22 14:35:01 CRON[143840]: pam_unix(cron:session): session closed for user root
Nov 22 14:39:12 sudo[145314]:     root : TTY=pts/0 ; PWD=/var/lib/algorand ; USER=algorand ; COMMAND=/usr/bin/nano -l config.json
Nov 22 14:39:12 sudo[145314]: pam_unix(sudo:session): session opened for user algorand by root(uid=0)
Nov 22 14:39:19 sudo[145314]: pam_unix(sudo:session): session closed for user algorand
Nov 22 14:39:25 sudo[145397]:     root : TTY=pts/0 ; PWD=/var/lib/algorand ; USER=root ; COMMAND=/usr/bin/systemctl restart algorand
Nov 22 14:39:25 sudo[145397]: pam_unix(sudo:session): session opened for user root by root(uid=0)
Nov 22 14:39:25 algod[138142]: Exiting on terminated
Nov 22 14:39:25 systemd[1]: Stopping Algorand daemon for mainnet in /var/lib/algorand...
Nov 22 14:39:43 systemd[1]: algorand.service: Succeeded.
Nov 22 14:39:43 systemd[1]: Stopped Algorand daemon for mainnet in /var/lib/algorand.
Nov 22 14:39:43 systemd[1]: Started Algorand daemon for mainnet in /var/lib/algorand.
Nov 22 14:39:43 sudo[145397]: pam_unix(sudo:session): session closed for user root
Nov 22 14:39:43 algod[145559]: Telemetry configured from '/var/lib/algorand/logging.config'
Nov 22 14:39:43 algod[145559]: Logging to:  /var/lib/algorand/node.log
Nov 22 14:39:49 sudo[145640]:     root : TTY=pts/0 ; PWD=/var/lib/algorand ; USER=root ; COMMAND=/usr/bin/systemctl status algorand
Nov 22 14:39:49 sudo[145640]: pam_unix(sudo:session): session opened for user root by root(uid=0)
Nov 22 14:39:49 sudo[145640]: pam_unix(sudo:session): session closed for user root
Nov 22 14:39:54 algod[145559]: couldn't initialize the node: reloadLedger.loadFromDisk tracker *ledger.catchpointTracker failed to loadFromDisk : resource table entries exceed the ones specified in the accountbase table
Nov 22 14:39:54 systemd[1]: algorand.service: Main process exited, code=exited, status=1/FAILURE
Nov 22 14:39:54 systemd[1]: algorand.service: Failed with result 'exit-code'.
Nov 22 14:39:59 systemd[1]: algorand.service: Scheduled restart job, restart counter is at 1.
Nov 22 14:39:59 systemd[1]: Stopped Algorand daemon for mainnet in /var/lib/algorand.
Nov 22 14:39:59 systemd[1]: Started Algorand daemon for mainnet in /var/lib/algorand.
Nov 22 14:39:59 algod[145703]: Telemetry configured from '/var/lib/algorand/logging.config'
Nov 22 14:39:59 algod[145703]: Logging to:  /var/lib/algorand/node.log
*** END SYSLOG *******

NOTE: Agree that the relevant log entry is "Nov 22 14:39:54 algod[145559]: couldn't initialize the node: reloadLedger.loadFromDisk tracker *ledger.catchpointTracker failed to loadFromDisk : resource table entries exceed the ones specified in the accountbase table"

$ goal node status
Cannot contact Algorand node: open /var/lib/algorand/algod.net: no such file or directory

**Comment by tom-gnade** [comment]
I tried to upload an extract of the node log, but it is 427MB. Do you have a location where I can sftp the file or drop it in a share somewhere? The first entry in the extract is "time":"2022-11-22T10:31:27.196992Z" so a few hours prior to the syslog extracts I provided above. I could try to trim down the log file some to see if I can get it closer to 14:20.

**Comment by onetechnical** [comment]
Hi @tom-gnade thanks, we've been able to dig deeper, and have already addressed this issue in the latest go-algorand (v3.12) release. We expect that to release real soon. Out of curiosity, what is the use case for this? Are you setting up a new relay? Do you synchronize the other catchpoints from a different relay?

**Comment by tom-gnade** [comment]
An archival relay node cannot use fast catchup to sync to the ledger. When we start up a relay, I do not enable catchpoint tracking or any other feature that might slow down our sync performance - we allow the relay to sync the ledger as rapidly as possible. When that is completed, I allow the relay to run in a steady-state mode for a while. The catchpoint tracking is really more of a network service that we would like to provide from our relay, it is not something that we would be consumers of. In this case, I wanted to enable catchpoint tracking so other consumers could use our relay to fast catchup to the ledger, as needed. I enabled catchpoint tracking only after the relay was fully synchronized (the old fashioned way, no fast catchup here!), and was running in a steady-state. The catchpoints consume a few hundred additional GB of storage, so I wanted to enable the feature and observe our performance and storage before and after the change, to determine if we wanted to keep the setting in place or not. Any relay that we would configure will be set up using this procedure. Catchpoint tracking would never be enabled during our initial sync, because it is network and performance overhead that detracts from our main goal, which is to synchronize to the ledger as fast as possible.

**Comment by onetechnical** [comment]
@tom-gnade Do you synchronize catchpoints from another server after catchup? Otherwise this would result in failed requests for catchpoints from relays because the files would be missing.

We believe this issue is fixed in v3.12.2-stable, can you confirm and close this if so?
